src/components/landing/pages/history/history.tsx

```diff
@@ -21,17 +21,36 @@ export const History: React.FC = () => {
   const [viewState, setViewState] = useState<HistoryToolbarState>(toolbarInitState)
 
   useEffect(() => {
-    const history = loadHistoryFromLocalStore()
-    setHistoryEntries(history)
+    refreshHistory()
   }, [])
 
   useEffect(() => {
-    if (historyEntries === []) {
+    if (!historyEntries || historyEntries === []) {
       return
     }
     setHistoryToLocalStore(historyEntries)
   }, [historyEntries])
 
+  const exportHistory = () => {
+    const data = 'data:text/json;charset=utf-8,' + encodeURIComponent(JSON.stringify(historyEntries))
```

Comment:
can you do that to get 'normal' json files?

Reply:
Ah this is not resolved yet.
It's completely possible. And would look like this:
```javascript
const data = 'data:text/json;charset=utf-8;base64,' + Buffer.from(JSON.stringify(dataObject)).toString('base64')
```

While I'm unsure if it's better to base64-encode the data rather than just replacing special characters, I've seen the base64 approach more often in the wild.

Reply:
I've done some quick tests with some example history exports and compared the size of the base64 result with the size of the uri-encoded result.

raw JSON: 2797 chars
uri-encoded: 4109 chars
base64: 3732 chars

While this is not such a big difference for this example, the heavy usage of {, } and : characters  in the JSON make the uri-encoded version much larger with larger histories.

Reply:
I worked that in